<?xml version="1.0"?>
<launch>

  <arg name="robot" default="mono_vio"/>
  <arg name="calibration_file"
    default="$(find msckf_mono)/config/camchain-imucam-euroc.yaml"/>

  <!-- Image Processor Nodelet  -->
  <group ns="$(arg robot)">
    <node pkg="nodelet" type="nodelet" name="image_processor"
      args="standalone msckf_mono/ImageProcessorNodelet"
      output="screen">

      <rosparam command="load" file="$(arg calibration_file)"/>
      <param name="grid_row" value="5"/>
      <param name="grid_col" value="6"/>
      <param name="grid_min_feature_num" value="5"/>
      <param name="grid_max_feature_num" value="7"/>
      <param name="pyramid_levels" value="3"/>
      <param name="patch_size" value="15"/>
      <param name="fast_threshold" value="10"/>
      <param name="max_iteration" value="50"/>
      <param name="track_precision" value="0.01"/>
      <param name="ransac_threshold" value="3"/>
      <param name="stereo_threshold" value="5"/>

      <remap from="~imu" to="/imu0"/>
      <remap from="~cam0_image" to="/cam0/image_raw"/>
      <remap from="~cam1_image" to="/cam1/image_raw"/>

    </node>
  </group>

</launch>
